tests/BookStore/Integration/Doctrine/DoctrineBookRepositoryTest.php

```diff
@@ -74,9 +76,9 @@ public function testOfId(): void
         static::assertEmpty($repository);
 
         $book = DummyBookFactory::createBook();
-        $repository->save($book);
-
-        static::getContainer()->get(EntityManagerInterface::class)->clear();
+        $repository->add($book);
+        self::$em->flush();
+        self::$em->clear();
```

Comment:
Just wondering, why do the clear is needed only here and not elsewhere?

Reply:
Idk, it was there before :)

Reply:
Probably because the new book entity is managed at that point, so it's in memory, and Doctrine will retrieve the entity directly instead of querying the database. By clearing, we force a new query to the database, but I'm just guessing that was the reason behind it.